Mathlib/Data/Matroid/Closure.lean

```diff
@@ -0,0 +1,207 @@
+/-
+Copyright (c) 2024 Peter Nelson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Peter Nelson
+-/
+import Mathlib.Data.Matroid.Basic
+
+/-!
+# Matroid Closure
+
+A `Flat` of a matroid `M` is a combinatorial analogue of a subspace of a vector space,
+and is defined to be a subset `F` of the ground set of `M` such that for each basis
+`I` for `M`, every set having `I` as a basis is contained in `F`.
+
+The *closure* of a set `X` in a matroid `M` is the intersection of all flats of `M` containing `X`.
+This is a combinatorial analogue of the linear span of a set of vectors.
+
+For `M : Matroid α`, this file defines a predicate `M.Flat : Set α → Prop` and a function
+`M.cl : Set α → Set α` corresponding to these notions, and develops API for the latter.
+API for `Matroid.Flat` will appear in another file; we include the definition here since
+it is used in the definition of `Matroid.cl`.
+
+## Main definitions
+
+* For `M : Matroid α` and `F : Set α`, `M.Flat F` means that `F` is a flat of `M`.
+* For `M : Matroid α` and `X : Set α`, `M.cl X` is the closure of `X` in `M`.
+
+## Implementation details
+
+If `X : Set α` satisfies `X ⊆ M.E`, then it is clear how `M.cl X` should be defined.
+But `M.cl X` needs to be defined for all `X : Set α`, so a decision is needed for how `M.cl`
+handles sets containing junk elements outside `M.E`. Ideally, such a choice should minimize
+the amount of keeping track of which sets are actually contained in the ground set.
+
+All such choices come with tradeoffs; the definition we opt for satisfies `M.cl X = M.cl (X ∩ M.E)`.
+The drawback here is that the statement `X ⊆ M.cl X` requires the assumption `X ⊆ M.E`.
+But all the other properties of closure work nicely without extra assumptions;
+closure is monotone, idempotent, and `M.cl X ⊆ M.E` for all `X`.
+-/
+
+open Set
+namespace Matroid
+
+variable {α ι : Type*} {M : Matroid α} {F I J X Y B C R : Set α} {e f x y : α}
```

Comment:
`B` and `C` are not used. Neither are `I` and `J`.

From the last part, only `{e : α}` is used.

Reply:
Thanks! This is because the PR is just the beginning of the repo file, which requires more in its full version.